stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -589,31 +720,44 @@ extension Unsafe${Mutable}BufferPointer {
     let count = slice.endIndex &- slice.startIndex
     self.init(start: base, count: count)
   }
+}
 
-  /// Deallocates the memory block previously allocated at this buffer pointer’s 
-  /// base address. 
+extension Unsafe${Mutable}BufferPointer where Element: ~Copyable {
+  /// Deallocates the memory block previously allocated at this buffer pointer’s
+  /// base address.
   ///
-  /// This buffer pointer's `baseAddress` must be `nil` or a pointer to a memory 
-  /// block previously returned by a Swift allocation method. If `baseAddress` is 
-  /// `nil`, this function does nothing. Otherwise, the memory must not be initialized 
-  /// or `Pointee` must be a trivial type. This buffer pointer's `count` must 
-  /// be equal to the originally allocated size of the memory block.
-  @inlinable // unsafe-performance
+  /// This buffer pointer's `baseAddress` must be `nil` or a pointer to a memory
+  /// block previously returned by a Swift allocation method. If `baseAddress`
+  /// is `nil`, this function does nothing. Otherwise, the memory must not be
+  /// initialized or `Pointee` must be a trivial type. This buffer pointer's
+  /// `count` must be equal to the originally allocated size of the memory
+  /// block.
+  @_alwaysEmitIntoClient
   public func deallocate() {
     _position?.deallocate()
   }
+}
 
-% if Mutable:
+extension Unsafe${Mutable}BufferPointer {
+  // TODO: Merge this back into the noncopyable variant once we have @_preInverseGenerics
```

Comment:
Are we going to merge this one back? Outdated comment?

Reply:
Yep, I forgot to merge it. Done!